frontend/appflowy_flutter/lib/plugins/document/application/document_share_bloc.dart

```diff
@@ -11,43 +11,82 @@ import 'package:freezed_annotation/freezed_annotation.dart';
 part 'document_share_bloc.freezed.dart';
 
 class DocumentShareBloc extends Bloc<DocumentShareEvent, DocumentShareState> {
-  DocumentShareBloc({required this.view})
-      : super(const DocumentShareState.initial()) {
-    on<ShareMarkdown>(_onShareMarkdown);
-  }
+  DocumentShareBloc({
+    required this.view,
+  }) : super(const DocumentShareState.initial()) {
+    on<DocumentShareEvent>((event, emit) async {
+      await event.when(
+        shareMarkdown: (path) async {
```

Comment:
Seems we can consolidate this into just one event, add the `DocumentExportType` as an event parameter.

Same with the `_saveMarkdownToPath` and `_saveHtmlToPath`, their only difference is the ExportType (Which you know now by the `DocumentExportType`.

Reply:
I removed the shareXXX event and replaced it with share(type, path?) instead.